sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java

```diff
@@ -374,159 +368,132 @@ public void deleteContact(String contactUuid) {
+		List<VisitSummaryExportDto> exportVisits = em.createQuery(cq).setFirstResult(first).setMaxResults(max).getResultList();
 
-		final Map<Long, List<ContactVisitsExportDto.ContactVisitsDetailsExportDto>> visitDetailsMap = new HashMap<>();
+		if (!exportVisits.isEmpty()) {
+			List<String> exportVisitUuids = exportVisits.stream().map(e -> e.getUuid()).collect(Collectors.toList());
 
-		final CriteriaQuery<Tuple> visitTupleQuery = cb.createQuery(Tuple.class);
-		final Root<Visit> visitRoot = visitTupleQuery.from(Visit.class);
+			CriteriaQuery<VisitSummaryExportDetails> visitsSq = cb.createQuery(VisitSummaryExportDetails.class);
+			Root<Contact> visitsSqRoot = visitsSq.from(Contact.class);
+			Join<Contact, Visit> visitsJoin = visitsSqRoot.join(Contact.VISITS, JoinType.LEFT);
+			Join<Visit, Symptoms> visitSymptomsJoin = visitsJoin.join(Visit.SYMPTOMS, JoinType.LEFT);
 
-		visitTupleQuery.multiselect(
-				visitRoot.get(Visit.VISIT_DATE_TIME).alias(Visit.VISIT_DATE_TIME),
-				visitRoot.get(Visit.VISIT_STATUS).alias(Visit.VISIT_STATUS),
-				visitRoot.get(Visit.SYMPTOMS).alias(Visit.SYMPTOMS),
-				visitRoot.get(Visit.PERSON).get(Person.ID).alias(Visit.PERSON));
-
-		if (!personIds.isEmpty()) {
-			visitTupleQuery.where(visitRoot.get(Visit.PERSON).get(Person.ID).in(personIds));
-		}
-
-		visitTupleQuery.orderBy(cb.asc(visitRoot.get(Visit.VISIT_DATE_TIME)));
+			visitsSq.where(ContactService.and(cb,
+					contactRoot.get(AbstractDomainObject.UUID).in(exportVisitUuids),
+					cb.isNotEmpty(visitsSqRoot.get(Contact.VISITS)))
+					);
+			visitsSq.multiselect(
+					visitsSqRoot.get(AbstractDomainObject.ID),
+					visitsJoin.get(Visit.VISIT_DATE_TIME),
+					visitsJoin.get(Visit.VISIT_STATUS),
+					visitSymptomsJoin
+					);
+			visitsSq.orderBy(cb.asc(visitsJoin.get(Visit.VISIT_DATE_TIME)));
 
-		final List<Tuple> tupleResult = em.createQuery(visitTupleQuery).getResultList();
+			List<VisitSummaryExportDetails> contactVisits = em.createQuery(visitsSq).getResultList();
 
-		for (Tuple tuple : tupleResult) {
-			final Date visitDateTime = extractTupleValue(tuple, Visit.VISIT_DATE_TIME);
-			final VisitStatus visitStatus = extractTupleValue(tuple, Visit.VISIT_STATUS);
-			final Symptoms symptoms = extractTupleValue(tuple, Visit.SYMPTOMS);
-			final Long personId = extractTupleValue(tuple, Visit.PERSON);
-			final ContactVisitsExportDto.ContactVisitsDetailsExportDto contactVisitsDetailsExportDto =
-					new ContactVisitsExportDto.ContactVisitsDetailsExportDto(visitDateTime, visitStatus,
-							symptoms.toHumanString(true, userLanguage));
-			if (visitDetailsMap.containsKey(personId)) {
-				visitDetailsMap.get(personId).add(contactVisitsDetailsExportDto);
-			} else {
-				final List<ContactVisitsExportDto.ContactVisitsDetailsExportDto> visitsDetails = new ArrayList<>();
-				visitsDetails.add(contactVisitsDetailsExportDto);
-				visitDetailsMap.put(personId, visitsDetails);
+			for (VisitSummaryExportDto exportVisit : exportVisits) {
```

Comment:
n*m - instead build Map of contact id + contact and then go once through the visits and put each into the correct contact.